### PR TITLE
[graphql/rpc] Auto create markdown doc from examples

### PR DESCRIPTION
**Pull request by oxade**
## Description 

As discussed with @stefan-mysten this starts the auto doc workflow
It could be prettier but it gets the ball rolling

## Test Plan 

Manual

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-graphql-rpc/src/examples.rs [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    // Generate the table of contents
+    for (id, group) in groups.iter().enumerate() {
+        let group_name = regularize_string(&group.name);
+        let group_name_toc = format!("[{}](#{})", group_name, id);
+        md.write(group_name_toc.heading(3)).unwrap();
+
+        for (inner, query) in group.queries.iter().enumerate() {
+            let inner_id = inner + 0xFFFF * id;
+            let inner_name = regularize_string(&query.name);
+
```
Comment (stefan-mysten) [inline crates/sui-graphql-rpc/src/examples.rs:104]: nit: spacing

File: crates/sui-graphql-rpc/src/examples.rs [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    // TODO: reduce multiple loops
+    // Generate the table of contents
+    for (id, group) in groups.iter().enumerate() {
+        let group_name = regularize_string(&group.name);
+        let group_name_toc = format!("[{}](#{})", group_name, id);
+        md.write(group_name_toc.heading(3)).unwrap();
+
+        for (inner, query) in group.queries.iter().enumerate() {
+            let inner_id = inner + 0xFFFF * id;
```
Comment (stefan-mysten) [inline crates/sui-graphql-rpc/src/examples.rs:102]: Don't really understand what's this doing.

**Review by stefan-mysten — COMMENTED** [review]
Overall this looks good, great work. If we could fix all those unwraps (I know, it's annoying), that'd be great.

File: crates/sui-graphql-rpc/src/examples.rs (hunk 2 above)
Comment (oxade) [reply crates/sui-graphql-rpc/src/examples.rs:102]: prevents table of content id collision

**Comment by oxade** [comment]
> Overall this looks good, great work. If we could fix all those unwraps (I know, it's annoying), that'd be great.

hmm since its mostly tests and not hot path code I think the unwraps are okay. Creating errors for this will be quite a task.
I'll do some cleanup but wont wanna over-invest in it

**Comment by oxade** [comment]
> Overall this looks good, great work. If we could fix all those unwraps (I know, it's annoying), that'd be great.



> > Overall this looks good, great work. If we could fix all those unwraps (I know, it's annoying), that'd be great.
> 
> hmm since its mostly tests and not hot path code I think the unwraps are okay. Creating errors for this will be quite a task. I'll do some cleanup but wont wanna over-invest in it

All done

File: crates/sui-graphql-rpc/src/examples.rs [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    let mut buf: PathBuf = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+    buf.push("examples");
+
+    let mut groups = vec![];
+    for entry in std::fs::read_dir(buf).map_err(|e| anyhow::anyhow!(e))? {
+        let entry = entry.map_err(|e| anyhow::anyhow!(e))?;
+        let path = entry.path();
+        let group_name = path
+            .file_stem()
+            .ok_or(anyhow::anyhow!("File stem cannot be read"))?
```
Comment (stefan-mysten) [inline crates/sui-graphql-rpc/src/examples.rs:50]:
```suggestion
            .ok_or(anyhow!("File stem cannot be read"))?
```

File: crates/sui-graphql-rpc/src/examples.rs [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+            .to_str()
+            .ok_or(anyhow::anyhow!("File stem cannot be read"))?
```
Comment (stefan-mysten) [inline crates/sui-graphql-rpc/src/examples.rs:52]:
```suggestion
            .ok_or(anyhow!("File stem cannot be read"))?
```

**Comment by stefan-mysten** [comment]
> > Overall this looks good, great work. If we could fix all those unwraps (I know, it's annoying), that'd be great.
> 
> > > Overall this looks good, great work. If we could fix all those unwraps (I know, it's annoying), that'd be great.
> > 
> > 
> > hmm since its mostly tests and not hot path code I think the unwraps are okay. Creating errors for this will be quite a task. I'll do some cleanup but wont wanna over-invest in it
> 
> All done

Lovely, there are still some `anyhow::anyhow!` that should be replaced with `anyhow!` as you already imported the macro.

**Review by stefan-mysten — APPROVED** [review]

**Pull request merged**